tests/e2e/e2e_garbage_collector_test.go

```diff
@@ -0,0 +1,135 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("Test garbage collection for unused envoy bootstrap config secrets",
+	OSMDescribeInfo{
+		Tier:   2,
```

Comment:
Reminder: Since this is Tier 2 it won't run as part of the PR CI, so make sure to keep testing this locally when you make changes.

(I know I forget to do this sometimes)